src/mono/mono/mini/aot-compiler.c

```diff
@@ -13886,12 +14022,12 @@ add_mibc_profile_methods (MonoAotCompile *acfg, char *filename)
 static void
 free_method_pinvoke_import_value (gpointer data)
 {
-	gchar **value = (gchar **)data;
+	PInvokeImportData *value = (PInvokeImportData *)data;
 	if (!value)
 		return;
-	g_free (value[0]);
-	g_free (value[1]);
-	g_free (value);
+	g_free ((char *)(value->module));
```

Comment:
Are these casts needed ?

Reply:
Not really needed, could be removed, or stay.